### PR TITLE
[BEAM-13229] [Website] bug side nav

### PR DESCRIPTION
**Pull request by bullet03**
- add refresh method to scrollspy on language-switch.js and page-nav.js
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by bullet03** [comment]
R: @nausharipov

**Review by aromanenko-dev — CHANGES_REQUESTED** [review]
Please, squash the commits and rebase it against `master` instead of merging `master` into the feature branch.

File: website/www/site/assets/js/language-switch-v2.js [hunk 1]
```diff
@@ -10,11 +10,26 @@
...
+    const elCurrentHeight = window.pageYOffset + clickedLangSwitchEl.getBoundingClientRect().top;
+    $('html, body').scrollTop(elCurrentHeight - elPreviousOffsetFromViewPort);
+}
+
 $(document).ready(function() {
     function Switcher(conf) {
         var id = conf["class-prefix"],
-            def = conf["default"],
-            langs = [];
+          def = conf["default"],
```
Comment (nausharipov) [inline website/www/site/assets/js/language-switch-v2.js:31]: please, format the file

File: website/www/site/assets/js/language-switch-v2.js [hunk 2]
```diff
@@ -89,7 +104,11 @@ $(document).ready(function() {
                 $("." + _self.wrapper + " ul li").click(function(el) {
                     // Making type preferences presistance, for user.
                     localStorage.setItem(_self.dbKey, $(this).data("type"));
+
+                    // Set scroll to new position because Safari and Firefox can't do it automatically, only Chrome under the good detects the correct position of viewport
```
Comment (nausharipov) [inline website/www/site/assets/js/language-switch-v2.js:108]: a typo: "Chrome under the good"

File: website/www/site/assets/js/language-switch-v2.js [hunk 3]
```diff
@@ -10,11 +10,26 @@
 // License for the specific language governing permissions and limitations under
 // the License.
 
+function getElPreviousPositionAndData(el) {
```
Comment (nausharipov) [inline website/www/site/assets/js/language-switch-v2.js:13]: let's change "el" to "element"

File: website/www/site/assets/js/language-switch-v2.js (hunk 2 above)
Comment (bullet03) [reply website/www/site/assets/js/language-switch-v2.js]: done

File: website/www/site/assets/js/language-switch-v2.js (hunk 3 above)
Comment (bullet03) [reply website/www/site/assets/js/language-switch-v2.js]: done

File: website/www/site/assets/js/language-switch-v2.js (hunk 1 above)
Comment (bullet03) [reply website/www/site/assets/js/language-switch-v2.js]: done

**Comment by bullet03** [comment]
We have cherry-picked and squashed commits from this branch into the new branch: https://github.com/apache/beam/pull/17731

**Pull request closed without merging**